tests/features/workspace.feature

```diff
@@ -0,0 +1,147 @@
+
+Scenario: Calling workspace services when workspace is started
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I run workspace start method
+    And   I call workspace services method
+    Then  I expect to receive service A and B with the following <property>s
+          |<property> |
+          |serviceName|
+          |displayName|
+          |proxy      |
+
+Scenario: Calling workspace components when workspace is started
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    When  I run workspace start method
+    And   I call workspace components method
+    Then  I expect to receive components 1 and 2 with the following <property>s
+          |<property>   |
+          |componentName|
+          |nodeSelector|
+          |reference   |
+
+Scenario: Calling workspace services when workspace is not started
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    And   Service A and service B includes a bootstrap that call registerService
+    When  I call workspace services method
+    Then  I expect to receive an error
+
+Scenario: Calling workspace components when workspace is not started
+    Given Workspace service instantiated with token 123
+    And   Configuration for token 123 that includes service A and B and components 1 and 2
+    When  I call workspace components method
+    Then  I expect to receive an error
+
+Scenario: Calling start method when workspace is started
+    Given Workspace service instantiated with token 123
```

Comment:
We should have 2 tests here :
1/ Workspace is started (meaning that I called `start` and it resolves) and I call start again
2/ Workspace is in starting process  (meaning that I called `start` and it didn't resolve or reject yet) and I call start again